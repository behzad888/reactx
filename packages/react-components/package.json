{
  "name": "react-components",
  "private": true,
  "description": "The standard React Components",
  "version": "0.0.1",
  "author": "Behzad Abbasi <overafaik.com>",
  "bugs": {
    "url": "https://github.com/reactx/reactx/issues"
  },
  "homepage": "https://github.com/reactx/reactx/tree/master/packages/react-components#readme",
  "keywords": [
    "reactx",
    "components"    
  ],
  "license": "MIT",
  "files": [
    "README.md",
    "alert.js"    
  ],
  "main": "index.js",
  "repository": "https://github.com/reactx/reactx/tree/master/packages/react-components"
}