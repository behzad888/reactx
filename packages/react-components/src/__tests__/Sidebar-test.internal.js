/**
 * Copyright (c) ReactX and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 *
 *
 */

'use strict';
import React from 'react';
import ReactDOM from 'react-dom';
import {Sidebar} from 'react-components';
import renderer from 'react-test-renderer';

describe('while running in a browser environment', () => {
  let container, wrapper;

  beforeEach(() => {
    jest.resetModules();

    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    ReactDOM.render(null, container);
    document.body.removeChild(container);
    container = null;
  });

  it('should render correctly', () => {
    wrapper = renderer
      .create(
        <div>
          <Sidebar>
            <p>salam</p>
          </Sidebar>
        </div>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
