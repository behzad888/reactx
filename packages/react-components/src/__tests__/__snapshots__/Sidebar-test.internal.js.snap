// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`while running in a browser environment should render correctly 1`] = `
<div>
  <div>
    <div
      style={
        Object {
          "WebkitTransform": "translateX(0%)",
          "WebkitTransition": "-webkit-transform .3s ease-out",
          "bottom": 0,
          "boxShadow": "2px 2px 4px rgba(0, 0, 0, 0.15)",
          "left": 0,
          "overflowY": "auto",
          "position": "absolute",
          "top": 0,
          "transform": "translateX(0%)",
          "transition": "transform .3s ease-out",
          "willChange": "transform",
          "zIndex": 2,
        }
      }
    >
      <div>
        <a
          href="#"
        >
          Mock menu item 
          1
        </a>
      </div>
    </div>
    <div
      onClick={[Function]}
      style={
        Object {
          "backgroundColor": "rgba(0,0,0,.3)",
          "bottom": 0,
          "left": 0,
          "opacity": 1,
          "position": "fixed",
          "right": 0,
          "top": 0,
          "transition": "opacity .3s ease-out, visibility .3s ease-out",
          "visibility": "visible",
          "zIndex": 1,
        }
      }
    />
    <div
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "bottom": 0,
          "left": 0,
          "overflowY": "auto",
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transition": "left .3s ease-out, right .3s ease-out",
        }
      }
    >
      <p>
        salam
      </p>
    </div>
  </div>
</div>
`;
